File: packages/types/src/template.ts
export type TemplateIndustry =
  | 'HOSPITAL'
  | 'PENSION'
  | 'STARTUP'
  | 'RESTAURANT'
  | 'SHOP'
  | 'EDUCATION'
  | 'GENERAL'

export interface CSSVariableSet {
  color_primary: string
  color_secondary: string
  color_accent: string
  color_background: string
  color_surface: string
  color_text_primary: string
  color_text_secondary: string

  font_heading: string
  font_body: string
  font_size_base: string
  font_weight_heading: string

  border_radius_base: string
  border_radius_card: string
  shadow_card: string

  spacing_section: string
  spacing_container: string
}

export interface SectionLayoutConfig {
  hero_style: 'fullscreen' | 'split' | 'centered' | 'magazine'
  card_style: 'grid' | 'list' | 'masonry' | 'carousel'
  nav_style: 'sticky' | 'transparent' | 'sidebar'
  footer_style: 'minimal' | 'full' | 'centered'
}

export interface Template {
  id: string
  name: string
  description: string
  thumbnail_url: string
  preview_url: string
  industry_tags: TemplateIndustry[]
  css_variables: CSSVariableSet
  section_layout: SectionLayoutConfig
  is_premium: boolean
  is_active: boolean
  sort_order: number
  created_at: string
}

export interface TemplateListItem {
  id: string
  name: string
  thumbnail_url: string
  preview_url: string
  industry_tags: TemplateIndustry[]
  is_premium: boolean
  is_current: boolean
}

export interface TenantTemplateOverride {
  id: string
  tenant_id: string
  template_id: string
  css_overrides: Partial<CSSVariableSet>
  applied_at: string
  updated_at: string
}

export interface TemplateApplyRequest {
  template_id: string
  preserve_content: true
}

export interface CSSOverrideUpdate {
  css_overrides: Partial<CSSVariableSet>
}

File: apps/backend/app/schemas/template.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class TemplateResponse(BaseModel):
    id: uuid.UUID
    template_type: str
    name: str
    description: str | None
    thumbnail_url: str | None
    css_variables: dict
    section_layouts: list
    is_active: bool
    min_plan: str

    model_config = {"from_attributes": True}


class TemplateListItem(TemplateResponse):
    """목록용: 현재 테넌트 플랜 기준 잠금 여부 포함."""

    locked: bool


class TemplateListResponse(BaseModel):
    templates: list[TemplateListItem]
    current_template_id: uuid.UUID | None


class TemplateApplyRequest(BaseModel):
    template_id: uuid.UUID


class TemplateCssOverrideUpdate(BaseModel):
    css_overrides: dict


class TenantTemplateOverrideResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    template_id: uuid.UUID
    css_overrides: dict
    applied_at: datetime
    template: TemplateResponse | None = None

    model_config = {"from_attributes": True}
